src/vsr/grid.zig

```diff
@@ -742,11 +835,18 @@ pub fn GridType(comptime Storage: type) type {
             }
 
             // Insert the block into the cache, and give the evicted block to `iop`.
-            const cache_index = grid.cache.insert_index(&read.address);
-            const cache_block = &grid.cache_blocks[cache_index];
-            std.mem.swap(BlockPtr, iop_block, cache_block);
-            @memset(iop_block.*, 0);
-            grid.cache_coherent.set(cache_index);
+            const cache_index =
+                if (read.cache_write) grid.cache.insert_index(&read.address) else null;
+            const block = block: {
+                if (read.cache_write) {
+                    const cache_block = &grid.cache_blocks[cache_index.?];
+                    std.mem.swap(BlockPtr, iop_block, cache_block);
+                    @memset(iop_block.*, 0);
```

Comment:
Unrelated, but this memset feels weird. First, we are doing memset here at all, and we generally try to avoid memsets. 

Second, this means that we have a spare block to memeset to begin with! I'd maybe expect for cached reads to always use the block from the cache, and for the uncached reads for the caller to provide the block.

Reply:
Right now we rely on blocks being zeroed in a bunch of places, even though there is only one where it really matters (constructing filter blocks). I'm planning on fixing all of that, probably as part of the grid block pool changes.

Swapping the iop block into the cache like this is nice since it means we don't have to "reserve" a slot in the cache for the duration of the read. It is worth reexamining that once we have a grid block pool, though.